bioinf_mega_tools.py

```diff
@@ -1,47 +1,84 @@
-from scripts.filter_params import is_filter_gc
-from scripts.filter_params import is_filter_length, is_filter_quality
-import scripts.dna_rna_tools
-import scripts.rw_fastq
+from abc import ABC
+from Bio import SeqIO
+from Bio.SeqUtils import GC
+import os, os.path
 
-operations = {
-    "transcribe": scripts.dna_rna_tools.transcribe,
-    "reverse": scripts.dna_rna_tools.reverse,
-    "complement": scripts.dna_rna_tools.complement,
-    "reverse_complement": scripts.dna_rna_tools.reverse_complement,
-}
 
+class BiologicalSequence(ABC):
+    def __init__(self, sequence: str):
+        self.sequence = sequence.upper()
 
-def run_dna_rna_tools(*args: str) -> str | dict:
-    """
-    The function returns DNA or RNA sequence
-    after one of the following operations:
-    - transcribe (works only for DNA)
-    - reverse
-    - complement
-    - reverse_complement (reverse and complement together)
+    def __len__(self):
+        return len(self.sequence)
 
-    Args:
-    last one - operation name
-    others - DNA/RNA sequence or sequences to operate
-    (if the value is not DNA/RNA returns a warning, not a sequence)
-    """
-    seqs = args[:-1]
-    operation = args[-1]
-    results = []
+    def __getitem__(self, index):
+        return self.sequence[index]
+
+    def __str__(self):
+        return self.sequence
+
+    def __repr__(self):
+        return f"{self.__class__.__name__}: {self.sequence}"
+
+    @staticmethod
+    def check_alphabet(sequence, alphabet):
+        return set(sequence).issubset(alphabet)
```

Comment:
Я бы не сказала, что это статик. Я бы скорее сказала, что alphabet это переменна на уровне класса (или `@property @abstractmethod` если совсем по уму), а
```suggestion
    def check_alphabet(slf):
        return set(self.sequence).issubset(self.alphabet)
```

Судя по коду и `complement_nucl` задумка то в ту сторону была, просто с проверкой алфавита реализация слегка странная получилась.